packages/desktop-client/src/components/reports/graphs/sankey-spreadsheet.tsx

```diff
@@ -0,0 +1,191 @@
+import q, { runQuery } from 'loot-core/src/client/query-helpers';
+import { send } from 'loot-core/src/platform/client/fetch';
+import { integerToAmount } from 'loot-core/src/shared/util';
+
+export default function createSpreadsheet(
+  start,
+  end,
+  categories,
+  conditions = [],
+  conditionsOp,
+) {
+  return async (spreadsheet, setData) => {
+    // gather filters user has set
+    let { filters } = await send('make-filters-from-conditions', {
+      conditions: conditions.filter(cond => !cond.customName),
+    });
+    const conditionsOpKey = conditionsOp === 'or' ? '$or' : '$and';
+
+    // create list of Income subcategories
+    const allIncomeSubcategories = [].concat(
+      ...categories
+        .filter(category => category.is_income === 1)
+        .map(category => category.categories),
+    );
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** ‏unnecessary array spread + concat.

```suggestion
    // create list of Income subcategories
    const allIncomeSubcategories = categories
        .filter(category => category.is_income === 1)
        .map(category => category.categories);
```